p2pd/metrics.go

```diff
@@ -0,0 +1,24 @@
+package main
+
+import (
+	psmetrics "github.com/libp2p/go-libp2p-pubsub/metrics"
+	"go.opencensus.io/exporter/prometheus"
+	"go.opencensus.io/stats/view"
+)
+
+func enableMetrics() (*prometheus.Exporter, error) {
+	opts := prometheus.Options{
+		Namespace: "libp2p_daemon",
+	}
+	pe, err := prometheus.NewExporter(opts)
+	if err != nil {
+		return nil, err
+	}
+	view.RegisterExporter(pe)
+
+	if err = psmetrics.Register(); err != nil {
```

Comment:
Per comments on the other PR, this would look like this: https://github.com/anacrolix/go-libp2p-dht-tool/blob/feat/oc-metrics/main.go#L377. What do you think?